### PR TITLE
feat: add 'apparmor' package

### PR DESCRIPTION
**Pull request by konrader**
Include AppArmor userland tool /sbin/apparmor_parser in rootfs to enable AppArmor support in containerd

As discussed in https://github.com/siderolabs/talos/issues/8715

### CODE REVIEW

**Comment by konrader** [comment]
This adds a 1MB userland tool `/sbin/apparmor_parser` in rootfs that `containerd` looks for as part of it's check for AppArmor support (https://github.com/containerd/containerd/blob/release/1.7/pkg/apparmor/apparmor_linux.go#L38)

It seem like apparmor LSM is not activated during Talos boot even if it's compiled into the kernel (after commit https://github.com/siderolabs/pkgs/commit/aefe000bd0cef8ec465b5fc14a1985ebba1fa429)
It can be activated with a kernel argument:
```
machine:
    install:
        extraKernelArgs:
            - security=apparmor
```
To verify that AppArmor LSM is enabled run `talosctl read /sys/kernel/security/lsm` that should return string that contains `apparmor` (amongst other LSM names)

**Comment by konrader** [comment]
I've been testing Talos with the new `apparmor` package and if it's included in Talos it's actually convenient that AppArmor LSM isn't activated by default (even if it's already compiled into kernel) since existing Talos users won't get surprised by default AppArmor profile getting applied to their Pod containers on a Talos upgrade.

**Scenario 1**: Running vanilla Talos
AppArmor built into kernel: `yes`
apparmor pkg: `yes`
AppArmor LSM activated: `no`
**=> containerd won't apply default AppArmor profile to started containers** (same behaviour as before)

**Scenario 2**: Running Talos with config extraKernelArgs: `security=apparmor`
AppArmor built into kernel: `yes`
apparmor pkg: `yes`
AppArmor LSM activated: `yes`
**=> containerd will apply default AppArmor profile to started containers and user can start Pods and specify their own AppArmor profile via security context**

This way anyone could run a normal Talos release and enable AppArmor support via machine config without having to make a custom Talos build but at the same time anyone upgrading won't be surprised by containerd automatically starting to add default AppArmor profile to their deployed Pods.

File: Pkgfile [hunk 1]
```diff
@@ -173,5 +173,10 @@ vars:
   zfs_version: 2.2.4
   zfs_sha256: 9790905f7683d41759418e1ef3432828c31116654ff040e91356ff1c21c31ec0
   zfs_sha512: 1d17e30573d594fb5c9ea77cde104616dca362fed7530296816d1b55173594f66170fcfb23ab57c27074f85b79d3eb557b4ee9a1c420e507b2434a7902d8dcc1
+
+  # renovate: datasource=git-refs versioning=git depName=https://gitlab.com/apparmor/apparmor.git
+  apparmor_version: v3.1.7
```
Comment (frezbo) [inline Pkgfile:178]: there seems to be v4.0.1 available, any reason an older one was picked?

Comment (konrader) [reply Pkgfile:178]: It seems like AppArmor 4.x is not "production ready" yet. When reading the release notes around 4.x they say both "release of the AppArmor that is in development" and "many of the features in apparmor 4 with require a development kernel" https://gitlab.com/apparmor/apparmor/-/releases/v4.0.1
Most Linux distributions that include AppArmor also package AppArmor 3.x so I was thinking of waiting bumping to AppArmor 4.1 until it comes out, which they indicate  "The following AppArmor 4.1 feature release is planned to be a regular release"

Comment (frezbo) [reply Pkgfile:178]: makes sense :+1:

**Comment by frezbo** [comment]
> I've been testing Talos with the new `apparmor` package and if it's included in Talos it's actually convenient that AppArmor LSM isn't activated by default (even if it's already compiled into kernel) since existing Talos users won't get surprised by default AppArmor profile getting applied to their Pod containers on a Talos upgrade.
> 
> **Scenario 1**: Running vanilla Talos AppArmor built into kernel: `yes` apparmor pkg: `yes` AppArmor LSM activated: `no` **=> containerd won't apply default AppArmor profile to started containers** (same behaviour as before)
> 
> **Scenario 2**: Running Talos with config extraKernelArgs: `security=apparmor` AppArmor built into kernel: `yes` apparmor pkg: `yes` AppArmor LSM activated: `yes` **=> containerd will apply default AppArmor profile to started containers and user can start Pods and specify their own AppArmor profile via security context**
> 
> This way anyone could run a normal Talos release and enable AppArmor support via machine config without having to make a custom Talos build but at the same time anyone upgrading won't be surprised by containerd automatically starting to add default AppArmor profile to their deployed Pods.

I think we can enable by default in 1.8, wdyt @smira ?

**Review by frezbo — APPROVED** [review]

**Comment by frezbo** [comment]
Probably a neat addition to this: https://github.com/siderolabs/talos/pull/5975

**Comment by smira** [comment]
> I think we can enable by default in 1.8, wdyt @smira ?

I feel we should document it _can_ be enabled for 1.8, and consider enabling it for 1.9 after we get some feedback, does it sound fair?

**Comment by frezbo** [comment]
> > I think we can enable by default in 1.8, wdyt @smira ?
> 
> I feel we should document it _can_ be enabled for 1.8, and consider enabling it for 1.9 after we get some feedback, does it sound fair?

cool, I have added a release notes when i merged the talos PR, let's push to 1.9 then

**Pull request merged**